roofit/roofitcore/src/RooAbsData.cxx

```diff
@@ -237,10 +238,9 @@ RooAbsData::~RooAbsData()
 
 void RooAbsData::convertToVectorStore()
 {
-  if (dynamic_cast<RooTreeDataStore*>(_dstore)) {
-    RooVectorDataStore* newStore =  new RooVectorDataStore(*(RooTreeDataStore*)_dstore,_vars,GetName()) ;
-    delete _dstore ;
-    _dstore = newStore ;
+   auto tmp = dynamic_cast<RooTreeDataStore *>(&*_dstore);
```

Comment:
for the first: thanks! will change

for the second: because i started with the unique_ptr business first. with the current (master branch) static defaultStorageType we cannot determine what storage type an actual instance is.